src/interpreter/ByteCodeInterpreter.cpp

```diff
@@ -2772,21 +2772,37 @@ NEVER_INLINE void ByteCodeInterpreter::initializeClassOperation(ExecutionState&
             name = AtomicString(state, code->m_name);
         }
 
+        Optional<Object*> outerClassConstructor;
+        auto home = state.mostNearestHomeObject();
+        if (home) {
+            if (home->isScriptClassConstructorPrototypeObject()) {
+                home = home->asScriptClassConstructorPrototypeObject()->constructor();
+            }
+            outerClassConstructor = home;
+        }
+
+
```

Comment:
This algorithm is very similar to the implementation of ```ExecutionState::findPrivateMemberContextObject```. Can you make it more efficient?

Reply:
Nice catch. I added some util function for that